internal/reconciliations/istio/installation.go

```diff
@@ -82,6 +84,11 @@ func (i *Installation) Reconcile(ctx context.Context, istioCR operatorv1alpha1.I
 			return istioCR, described_errors.NewDescribedError(err, "Could not get configuration from Istio Operator file")
 		}
 
+		ingressGatewayNeedsRestart, err := IngressGatewayNeedsRestart(istioCR)
```

Comment:
Right now the invocation of ` IngressGatewayNeedsRestart` and `RestartIngressGateway` are two separate function calls that are also a little bit separated in the code. I think now hat we rely on lastAppliedConfiguration instead of the CM on the cluster, we can also encapsulate the IngressGatewayNeedsRestart in the function doing the IGW restart. This will make the code in the installation process easier and we can have the logic about the ingress gateway restart more encapsulated.